servers/bitwindow/engines/bitcoind_engine.go

```diff
@@ -0,0 +1,257 @@
+package bitcoind_engine
+
+import (
+	"context"
+	"database/sql"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"time"
+
+	"connectrpc.com/connect"
+	corepb "github.com/barebitcoin/btc-buf/gen/bitcoin/bitcoind/v1alpha"
+	coreproxy "github.com/barebitcoin/btc-buf/server"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/rs/zerolog"
+)
+
+func New(
+	bitcoind *coreproxy.Bitcoind,
+	db *sql.DB,
+) *Parser {
+	return &Parser{
+		bitcoind: bitcoind,
+		db:       db,
+	}
+}
+
+// Parser is responsible for parsing blocks from bitcoind and storing OP_RETURN data in SQLite
+type Parser struct {
+	bitcoind *coreproxy.Bitcoind
+	db       *sql.DB
+}
+
+// Run runs the engine. It checks if a new block has been mined,
+// and if so, handles it!
+//
+// Should be started in a goroutine.
+func (p *Parser) Run(ctx context.Context) error {
+	// Check every other second. We wouldn't wanna miss a nice rate, now would we
+	alertTicker := time.NewTicker(2 * time.Second)
+	defer alertTicker.Stop()
+
+	zerolog.Ctx(ctx).Info().
+		Msgf("bitcoind_engine/parser: starting parser ticker")
+
+	processing := false
+	for {
+		select {
+		case <-ctx.Done():
+			zerolog.Ctx(ctx).Info().
+				Msgf("bitcoind_engine/parser: stopping parser ticker")
+			return nil
+
+		case <-alertTicker.C:
+			if processing {
+				continue
+			}
+
+			processing = true
+			p.handleTick(ctx)
+			processing = false
+		}
+	}
+}
+
+func (p *Parser) handleTick(ctx context.Context) {
```

Comment:
this method should really return an error, and then the consumer can handle that error (i.e. log it)